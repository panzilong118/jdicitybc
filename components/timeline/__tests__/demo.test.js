import * as React from "react";

import renderer from 'react-test-renderer';

import Basic from '../demo/Basic';

import Color from '../demo/Color';

import Custom from '../demo/Custom';

import Pending from '../demo/Pending';

describe('<Timeline />', () => {

    it('Basic usage of Timeline', () => {
        const demo = renderer.create(<Basic />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('Color usage of Timeline', () => {
        const demo = renderer.create(<Color />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('Custom usage of Timeline', () => {
        const demo = renderer.create(<Custom />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('Pending usage of Timeline', () => {
        const demo = renderer.create(<Pending />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

});
