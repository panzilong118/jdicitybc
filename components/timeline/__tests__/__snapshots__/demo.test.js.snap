// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timeline /> Basic usage of Timeline 1`] = `
<ul
  className="jc-timeline"
>
  <li
    className="jc-timeline-item"
  >
    <div
      className="jc-timeline-item-tail"
    />
    <div
      className="jc-timeline-item-head jc-timeline-item-head-blue"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="jc-timeline-item-content"
    >
      Create a services site 2015-09-01
    </div>
  </li>
  <li
    className="jc-timeline-item"
  >
    <div
      className="jc-timeline-item-tail"
    />
    <div
      className="jc-timeline-item-head jc-timeline-item-head-blue"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="jc-timeline-item-content"
    >
      Solve initial network problems 2015-09-01
    </div>
  </li>
  <li
    className="jc-timeline-item"
  >
    <div
      className="jc-timeline-item-tail"
    />
    <div
      className="jc-timeline-item-head jc-timeline-item-head-blue"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="jc-timeline-item-content"
    >
      Technical testing 2015-09-01
    </div>
  </li>
  <li
    className="jc-timeline-item jc-timeline-item-last"
  >
    <div
      className="jc-timeline-item-tail"
    />
    <div
      className="jc-timeline-item-head jc-timeline-item-head-blue"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="jc-timeline-item-content"
    >
      Network problems being solved 2015-09-01
    </div>
  </li>
</ul>
`;

exports[`<Timeline /> Color usage of Timeline 1`] = `
<ul
  className="jc-timeline"
>
  <li
    className="jc-timeline-item"
  >
    <div
      className="jc-timeline-item-tail"
    />
    <div
      className="jc-timeline-item-head jc-timeline-item-head-green"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="jc-timeline-item-content"
    >
      Create a services site 2015-09-01
    </div>
  </li>
  <li
    className="jc-timeline-item"
  >
    <div
      className="jc-timeline-item-tail"
    />
    <div
      className="jc-timeline-item-head jc-timeline-item-head-green"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="jc-timeline-item-content"
    >
      Create a services site 2015-09-01
    </div>
  </li>
  <li
    className="jc-timeline-item"
  >
    <div
      className="jc-timeline-item-tail"
    />
    <div
      className="jc-timeline-item-head jc-timeline-item-head-red"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="jc-timeline-item-content"
    >
      <p>
        Solve initial network problems 1
      </p>
      <p>
        Solve initial network problems 2
      </p>
      <p>
        Solve initial network problems 3 2015-09-01
      </p>
    </div>
  </li>
  <li
    className="jc-timeline-item jc-timeline-item-last"
  >
    <div
      className="jc-timeline-item-tail"
    />
    <div
      className="jc-timeline-item-head jc-timeline-item-head-blue"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="jc-timeline-item-content"
    >
      <p>
        Technical testing 1
      </p>
      <p>
        Technical testing 2
      </p>
      <p>
        Technical testing 3 2015-09-01
      </p>
    </div>
  </li>
</ul>
`;

exports[`<Timeline /> Custom usage of Timeline 1`] = `
<ul
  className="jc-timeline"
>
  <li
    className="jc-timeline-item"
  >
    <div
      className="jc-timeline-item-tail"
    />
    <div
      className="jc-timeline-item-head jc-timeline-item-head-blue"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="jc-timeline-item-content"
    >
      Create a services site 2015-09-01
    </div>
  </li>
  <li
    className="jc-timeline-item"
  >
    <div
      className="jc-timeline-item-tail"
    />
    <div
      className="jc-timeline-item-head jc-timeline-item-head-blue"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="jc-timeline-item-content"
    >
      Solve initial network problems 2015-09-01
    </div>
  </li>
  <li
    className="jc-timeline-item"
  >
    <div
      className="jc-timeline-item-tail"
    />
    <div
      className="jc-timeline-item-head jc-timeline-item-head-custom jc-timeline-item-head-red"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    >
      <i
        className="jcicon jcicon-clock-circle-o "
        style={
          Object {
            "fontSize": "16px",
          }
        }
      />
    </div>
    <div
      className="jc-timeline-item-content"
    >
      Technical testing 2015-09-01
    </div>
  </li>
  <li
    className="jc-timeline-item jc-timeline-item-last"
  >
    <div
      className="jc-timeline-item-tail"
    />
    <div
      className="jc-timeline-item-head jc-timeline-item-head-blue"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="jc-timeline-item-content"
    >
      Network problems being solved 2015-09-01
    </div>
  </li>
</ul>
`;

exports[`<Timeline /> Pending usage of Timeline 1`] = `
<ul
  className="jc-timeline jc-timeline-pending"
>
  <li
    className="jc-timeline-item"
  >
    <div
      className="jc-timeline-item-tail"
    />
    <div
      className="jc-timeline-item-head jc-timeline-item-head-blue"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="jc-timeline-item-content"
    >
      Create a services site 2015-09-01
    </div>
  </li>
  <li
    className="jc-timeline-item"
  >
    <div
      className="jc-timeline-item-tail"
    />
    <div
      className="jc-timeline-item-head jc-timeline-item-head-blue"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="jc-timeline-item-content"
    >
      Solve initial network problems 2015-09-01
    </div>
  </li>
  <li
    className="jc-timeline-item jc-timeline-item-last"
  >
    <div
      className="jc-timeline-item-tail"
    />
    <div
      className="jc-timeline-item-head jc-timeline-item-head-blue"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="jc-timeline-item-content"
    >
      Technical testing 2015-09-01
    </div>
  </li>
  <li
    className="jc-timeline-item jc-timeline-item-pending"
  >
    <div
      className="jc-timeline-item-tail"
    />
    <div
      className="jc-timeline-item-head jc-timeline-item-head-custom jc-timeline-item-head-blue"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    >
      <i
        className="jcicon jcicon-loading  jcicon-spin"
        style={Object {}}
      />
    </div>
    <div
      className="jc-timeline-item-content"
    >
      Recording...
    </div>
  </li>
</ul>
`;
